### PR TITLE
UCT/MM: Enlarge receive descriptor chunks to avoid shm id overflow.

### LINKED ISSUE #2078: failure on mpi_test_suite on 32x32, shared memory segment count reached
```
mpirun  -mca coll ^hcoll -mca pml ucx -mca btl self -x UCX_NET_DEVICES=mlx5_0:1 --map-by node /hpc/scrap/users/yosefe/workspace/ompi-tests/mpi_test_suite/mpi_test_suite -t 'All,^one-sided,^IO
...
Collective tests Bcast (31/48), comm Duplicated MPI_COMM_WORLD (4/14), type MPI_INDEXED_INT (24/28)
Collective tests Bcast (31/48), comm Duplicated MPI_COMM_WORLD (4/14), type MPI_HINDEXED_INT (25/28)
Collective tests Bcast (31/48), comm Duplicated MPI_COMM_WORLD (4/14), type MPI_STRUCT_INT (26/28)
Collective tests Bcast (31/48), comm Duplicated MPI_COMM_WORLD (4/14), type MPI_TYPE_MIX (27/28)
Collective tests Bcast (31/48), comm Duplicated MPI_COMM_WORLD (4/14), type MPI_TYPE_MIX_ARRAY (28/28)
Collective tests Bcast (31/48), comm Reversed MPI_COMM_WORLD (5/14), type MPI_CHAR (1/28)
[clx-hercules-062:5330 :0] Process frozen...
==== backtrace ====
[clx-hercules-062:5332 :0]     mm_sysv.c:68   Fatal: failed to allocate 1065072 bytes with mm
[1513201433.156702] [clx-hercules-062:5332 :0]            sys.c:482  UCX  ERROR shmget failed: No space left on device. (size=1069056). The max number of shared memory segments in the system is = 4096. Please try to increase this value through /proc/sys/kernel/shmmni
...
 0 0x0000000000050f51 uct_sysv_alloc()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/uct/sm/mm/mm_sysv.c:68
 1 0x00000000000509eb uct_mm_mem_alloc()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/uct/sm/mm/mm_md.c:38
 2 0x000000000001c157 uct_mem_alloc()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/uct/base/uct_mem.c:116
 3 0x000000000001c53b uct_iface_mem_alloc()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/uct/base/uct_mem.c:264
 4 0x000000000001c613 uct_iface_mp_chunk_alloc()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/uct/base/uct_mem.c:329
 5 0x000000000003fc59 ucs_mpool_grow()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/ucs/datastruct/mpool.c:173
 6 0x000000000003fdf0 ucs_mpool_get_grow()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/ucs/datastruct/mpool.c:221
 7 0x000000000004ec02 ucs_mpool_get_inline()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/ucs/datastruct/mpool.inl:23
 8 0x000000000004ec02 uct_mm_iface_progress()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/uct/sm/mm/mm_iface.c:258
 9 0x0000000000013d82 ucs_callbackq_dispatch()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/ucs/datastruct/callbackq.h:166
10 0x0000000000013d82 uct_worker_progress()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/uct/api/uct.h:1623
11 0x0000000000013d82 ucp_worker_progress()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ucx-master/src/ucp/core/ucp_worker.c:1180
12 0x0000000000003377 mca_pml_ucx_progress()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/ompi/mca/pml/ucx/pml_ucx.c:451
13 0x000000000003164c opal_progress()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/opal/runtime/opal_progress.c:222
14 0x0000000000047d25 ompi_request_wait_completion()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/ompi/../ompi/request/request.h:392
15 0x0000000000047d25 ompi_request_default_wait()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/ompi/request/req_wait.c:42
16 0x0000000000005983 mca_pml_ucx_send()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/ompi/mca/pml/ucx/pml_ucx.c:747
17 0x000000000007b375 ompi_coll_base_bcast_intra_split_bintree()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/ompi/mca/coll/base/coll_base_bcast.c:483
18 0x000000000000499c ompi_coll_tuned_bcast_intra_dec_fixed()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/ompi/mca/coll/tuned/coll_tuned_decision_fixed.c:265
19 0x000000000005b1d1 PMPI_Bcast()  /scrap/jenkins/scrap/workspace/hpc-power-pack/label/hpc-test-node/hpcx_root/src/hpcx-v2.0.0-gcc-MLNX_OFED_LINUX-4.2-1.0.0.0-redhat7.2-x86_64/ompi-v3.0.x/ompi/mpi/c/profile/pbcast.c:111
20 0x0000000000407fa4 tst_coll_bcast_run()  /hpc/scrap/users/yosefe/workspace/ompi-tests/mpi_test_suite/coll/tst_coll_bcast.c:72
21 0x0000000000458457 tst_test_run_func()  /hpc/scrap/users/yosefe/workspace/ompi-tests/mpi_test_suite/tst_tests.c:1461
22 0x0000000000434389 main()  /hpc/scrap/users/yosefe/workspace/ompi-tests/mpi_test_suite/mpi_test_suite.c:637
23 0x0000000000021c05 __libc_start_main()  ???:0
24 0x00000000004052e9 _start()  ???:0
===================
[clx-hercules-062:5328 :0] Process frozen...
```

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2078

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5533/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:bgate:retest

File: src/uct/sm/mm/mm_iface.c [hunk 1]
```diff
@@ -520,14 +520,16 @@ static UCS_CLASS_INIT_FUNC(uct_mm_iface_t, uct_md_h md, uct_worker_h worker,
...
-                                  mm_config->fifo_size * 2,
+                                  512,
                                   uct_mm_iface_recv_desc_init,
                                   "mm_recv_desc");
     if (status != UCS_OK) {
         ucs_error("Failed to create a receive descriptor memory pool for the MM transport");
         goto err_close_signal_fd;
     }
 
+    ucs_mpool_grow(&self->recv_desc_mp, mm_config->fifo_size * 2);
```
Comment (alinask) [inline src/uct/sm/mm/mm_iface.c]: why need to make the mpool grow here?

Comment (yosefe) [reply src/uct/sm/mm/mm_iface.c]: so the first allocated chunk would be small, to reduce initial memory usage

**Pull request merged**